File: parent335/pinyougou-interface/src/main/java/cn/itcast/core/service/address/AddressService.java
package cn.itcast.core.service.address;

import cn.itcast.core.pojo.address.Address;
import cn.itcast.core.pojo.address.Areas;
import cn.itcast.core.pojo.address.Cities;
import cn.itcast.core.pojo.address.Provinces;

import java.util.List;

public interface AddressService {
    /**
     * 获取当前登录人的地址列表
     */
    public List<Address> findAddressList(String userId);

    /**
     * c
     *
     */
    public List<Address> findAddressListByLoginUser(String userName);

    Address findOne(Long id);

    void add(Address address);

    void update(Address address);

    void delete(Long id);

    public void setDefaultAddress(Long id);

    public List<Provinces> findProvinceByparentId(String parentId);

    public List<Cities> findCityByparentId(String parentId);

    public List<Areas> findAreaByparentId(String parentId);
//
}

File: parent335/pinyougou-web-user/src/main/webapp/js/service/addressService.js
// 地址服务层
app.service('addressService', function($http) {

    // 获取地址列表
    this.findAddressList = function(){
        return $http.get('address/findListByLoginUser.do');
    };
    // 增加
    this.add = function(address){
        return $http.post('address/add.do/', address);
    };
    // 修改
    this.update = function(address){
        return $http.post('address/update.do/', address);
    };
    this.findOne = function (id) {
        return $http.get('address/findOne.do?id='+id);
    };
    this.del = function (id) {
        return $http.get('address/delete.do?id='+id);
    };

    //设置默认地址
    this.setDefaultAddress = function (id) {
        return $http.get('address/setDefaultAddress.do?id=' + id);
    }

    this.findProvince = function(parentId){
        return $http.get("../address/findProvince.do?parentId="+parentId);
    }
    this.findCity = function(parentId){
        return $http.get("../address/findCity.do?parentId="+parentId);
    }
    this.findArea = function(parentId){
        return $http.get("../address/findArea.do?parentId="+parentId);
    }

});
